lib/libutee/include/utee_defines.h

```diff
@@ -46,6 +46,7 @@
 #define TEE_MAIN_ALGO_X25519     0x44 /* Not in v1.2 spec */
 #define TEE_MAIN_ALGO_SHAKE128   0xC3 /* OP-TEE extension */
 #define TEE_MAIN_ALGO_SHAKE256   0xC4 /* OP-TEE extension */
+#define TEE_MAIN_ALGO_X448	0xC5 /* OP-TEE extension */
```

Comment:
If 0xC5 is aligned with 0xC4, a space is required. In this case, the CI/Code style may fail.

Reply:
> If 0xC5 is aligned with 0xC4, a space is required. 

Yes

> In this case, the CI/Code style may fail.

No

Reply:
We aligned 0xC5 with 0xC4 on the line above.

Reply:
We run testcases and found that  If the TEE_MAIN_ALGO_X448 value is 0xC5, when call `TEE_ObjectType key_type = TEE_ALG_GET_KEY_TYPE(algorithm, with_private_key)` in tee_svc_cryp.c, `key_type = 0xA0000000 | 0xC5 | 0x01000000 = 0xA10000C5` is not match with `TEE_TYPE_X448_KEYPAIR = 0xA1000049`, which lead to allocate oobj failed then testcase running failed. 
So we think if the TEE_MAIN_ALGO_X448 macro value should be defined as 0x49? @jenswi-linaro @jforissier

Reply:
Possibly. If you want to change that, we need two things:
1. Add support for X448 in optee_os
2. Add a test case in optee_test

Reply:
Since there are no support for X448 in crypto library，so we run test cases through hardware deployment.  The running results are also obtained by this way. @jforissier

Reply:
@loubaihui I understand that. It is difficult to review patches when we don't have the full code, but I think your proposal to change the value to 0x49 is OK. Can you please open a PR?

Reply:
I'm sorry, I don't follow why it should be 0x49. Doesn't that conflict with the standard?

Reply:
> I'm sorry, I don't follow why it should be 0x49. Doesn't that conflict with the standard?

The standard does not define `TEE_MAIN_ALGO_X448`, in fact there is no `TEE_MAIN_ALGO_*` anymore. So we can chose any value we want (and we should probably remove all these constants from the public API -- to do for 4.0?). 0x49 is the value needed so that `TEE_ALG_GET_KEY_TYPE(TEE_ALG_X448, true) == TEE_TYPE_X448_KEYPAIR` and `TEE_ALG_GET_KEY_TYPE(TEE_ALG_X448, false) == TEE_TYPE_X448_PUBLIC_KEY`.

Reply:
OK, thanks for the explanation. Yes, some cleanup seems to be needed.

Reply:
> @loubaihui I understand that. It is difficult to review patches when we don't have the full code, but I think your proposal to change the value to 0x49 is OK. Can you please open a PR?

I'm sorry for my negligence. we will open a pr for the modification.

Reply:
> @loubaihui I understand that. It is difficult to review patches when we don't have the full code, but I think your proposal to change the value to 0x49 is OK. Can you please open a PR?

I'm sorry for my negligence. We will open a pr for the modification.

> @loubaihui I understand that. It is difficult to review patches when we don't have the full code, but I think your proposal to change the value to 0x49 is OK. Can you please open a PR?

I'm sorry for my negligence. We will open a pr for the modification.